### ISSUE TITLE
Asking a question? PLEASE READ FIRST

### ISSUE THREAD

**Issue by Jeffail** [benthosdev/benthos#2026]
TLDR; please do not open new issues in order to ask a question, either raise a new [discussion](https://github.com/benthosdev/benthos/discussions) or join one of our [community spaces](https://www.benthos.dev/community/). In an ideal world I would have continued to answer questions where ever they may appear, but given the scope of the project at this point it's becoming increasingly important that we're able to separate questions from other tasks in our alerting and triaging processes.

---

Our ability to ask questions defines who we are as humans. Without fire we would be apes, but without questions we would be seashells or something. When a dog tilts it's head to ask "do you love me?" it is cutely reminding us that dogs are ready to replace humans at any time.

Ever since Benthos was a young pre-v1 whippersnapper I've attempted to triage questions with the same urgency as enhancement and bug issues. Whenever a question popped up I would traditionally read and understand it as soon as I received the notification, and following that I would either answer it there and then or make a mental note to have an answer put together as soon as my game of power-walk-water-balloon-tennis was over.

My reasoning behind this was that answering questions is often a good source of discovering new enhancement requests (and also bugs with usability, documentation, etc). Extracting these discoveries often requires a bit of further discussion with the question asker beyond just getting them an answer, and the quicker you're able to engage with them the more useful that discussion will be. Therefore, treating questions with the same urgency as other issues was a bit of a cheat code for project development, and since people who ask questions generally enjoy quick answers everyone comes out a winner.

I would love to carry on with a philosophy of treating questions as first-class citizens, and a small part of that is allowing you to ask questions where ever it suits you. See me in a public toilet? I'm fully prepared to answer that question for you there and then. As it stands we currently have a few [online community spaces](https://www.benthos.dev/community) where you can pop questions as well as the Github issues and discussions tabs, and all of these spaces are growing fast.

It's long been apparent that for many people asking questions in a new Github issue is their first intuition. Github issues are great for searchability and for keeping track of a long discussion, but that also means they exist in the same notification pool as other issues. Another annoyance with Github issues from our perspective is that in order to isolate them from "real" issues (enhancements and bugs) we need to go through the process of tagging each one, and periodically close stale ones.

These annoyances are minor, but as maintainers of a rapidly growing project our attention economy is becoming increasingly strained and we're steadily losing more and more TikTok dances from our repertoire. The need for a rebalance is therefore upon us. I am making the decision to politely discourage new questions from being asked in the form of an issue. From now on please either ask your question as a new [discussion](https://github.com/benthosdev/benthos/discussions) or join us in one of our [community spaces](https://www.benthos.dev/community).

There will be no punishment for anyone failing this. Sometimes the lines between a question and bug aren't apparent until a bit of discussion has been had, so please open an issue if you're uncertain. However, please also don't be upset if we choose to close your question issue with a link to this one.
